### PR TITLE
Api updates 3.5.4

### PR DESCRIPTION
**Pull request by gabrielalao**
- expose `chainMetadata` param to `searchPlaces` function
- expose `scheduledArrivalAt` param to `startTrip`

### CODE REVIEW

File: android/src/main/java/io/radar/react/RNRadarModule.java [hunk 1]
```diff
@@ -578,6 +579,7 @@ public void searchPlaces(ReadableMap optionsMap, final Promise promise) {
         }
         int radius = optionsMap.hasKey("radius") ? optionsMap.getInt("radius") : 1000;
         String[] chains = optionsMap.hasKey("chains") ? RNRadarUtils.stringArrayForArray(optionsMap.getArray("chains")) : null;
+        Map<String, String> chainMetadata = optionsMap.hasKey("chainMetadata") ? RNRadarUtils.stringStringMap(optionsMap.getMap("chainMetadata")) : null;
```
Comment (jrtibbetts) [inline android/src/main/java/io/radar/react/RNRadarModule.java:582]: If `stringStringMap()` returns `null` when the value passed in is `null`, do you even need the `hasKey()` check?

File: ios/RNRadar.m [hunk 2]
```diff
@@ -321,6 +321,9 @@ - (void)didLogMessage:(NSString *)message {
 
 RCT_EXPORT_METHOD(startTrip:(NSDictionary *)optionsDict resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject) {
     RadarTripOptions *options = [RadarTripOptions tripOptionsFromDictionary:optionsDict];
+    if (options.scheduledArrivalAt) {
+        options.scheduledArrivalAt = [RCTConvert NSDate:options.scheduledArrivalAt];
```
Comment (jrtibbetts) [inline ios/RNRadar.m:325]: What does this do?

**Review by jrtibbetts — APPROVED** [review]

File: ios/RNRadar.m (hunk 2 above)
Comment (gabrielalao) [reply ios/RNRadar.m:325]: React Native Bridge can't deliver Date object itself, so we need to deliver Date value (Number) and on iOS we need to convert to NSDate value.

File: android/src/main/java/io/radar/react/RNRadarModule.java (hunk 1 above)
Comment (gabrielalao) [reply android/src/main/java/io/radar/react/RNRadarModule.java:582]: ok i will update

File: ios/RNRadar.m (hunk 2 above)
Comment (jrtibbetts) [reply ios/RNRadar.m:325]: Perfect. Got it.

**Comment by tjulien** [comment]
@gabrielalao Can you add tests for the new function here: https://github.com/radarlabs/react-native-radar/blob/master/test/index.test.js

File: android/src/main/java/io/radar/react/RNRadarModule.java [hunk 3]
```diff
@@ -78,16 +79,54 @@ public void setUserId(String userId) {
         Radar.setUserId(userId);
     }
 
+    @ReactMethod
+    public void getUserId(final Promise promise) {
```
Comment (tjulien) [inline android/src/main/java/io/radar/react/RNRadarModule.java:83]: Can you add test for all of the new functions here? https://github.com/radarlabs/react-native-radar/blob/master/test/index.test.js

File: android/src/main/java/io/radar/react/RNRadarModule.java [hunk 4]
```diff
@@ -146,8 +185,20 @@ public void requestPermissions(boolean background, final Promise promise) {
     }
 
     @ReactMethod
-    public void getLocation(final Promise promise) {
-        Radar.getLocation(new Radar.RadarLocationCallback() {
+    public void getLocation(String desiredAccuracy, final Promise promise) {
+        
+        RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy accuracyLevel = RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.MEDIUM;
```
Comment (tjulien) [inline android/src/main/java/io/radar/react/RNRadarModule.java:190]: We can't default to medium confidence.  When `desiredAccuracy` is null we should call the version of the `Radar.getLocation()` function that doesn't take a `desiredAccuracy` param.

And it `desiredAccuracy` is a string but not a known value (high, medium, low) we should just promise.reject() with an error message

Comment (gabrielalao) [reply android/src/main/java/io/radar/react/RNRadarModule.java:190]: @tjulien Got it. I will add test for those new functions and update 'getLocation' function.

**Comment by tjulien** [comment]
@gabrielalao Looks like both the iOS and Android build's are failing. The iOS build I think is an intermittent failure, the Android one I have seen before.

If you can get those to pass and the bump the version number (steps [here](https://www.notion.so/radarlabs/How-to-release-the-React-Native-SDK-21c70f16e0bd4fe28425923645cadda0)), then I think we're good to merge this and create a release.

**Comment by gabrielalao** [comment]
@tjulien I will check once more both android and ios and bump up the version and release.

**Comment by tjulien** [comment]
@gabrielalao Once the two builds are passing I'll approve this PR.  Then you can merge it to `master` and do a Github Release.

**Comment by gabrielalao** [comment]
@tjulien I upgrade expo sdk 40 to 43 and the android build goes well, but due to react-native-cli issue, ios build did not success. But on my pc, all builds works well.

**Comment by tjulien** [comment]
@gabrielalao Ok, so you're still working on the iOS build?

**Comment by gabrielalao** [comment]
@tjulien Just finished everything on ios, only needs to bump up version only.

**Comment by gabrielalao** [comment]
@tjulien Everything is ready to release.

File: android/build.gradle [hunk 5]
```diff
@@ -18,7 +18,7 @@ android {
         minSdkVersion 16
         targetSdkVersion 31
         versionCode 1
-        versionName '3.5.6'
+        versionName '3.5.7'
```
Comment (tjulien) [inline android/build.gradle:21]: The latest release for Android is 3.5.6

**Comment by tjulien** [comment]
@gabrielalao Looks like the iOS build is still failing

File: android/build.gradle (hunk 5 above)
Comment (gabrielalao) [reply android/build.gradle:21]: this version code is for react native sdk, not android sdk version. do you want to leave the version code 3.5.6?

**Comment by gabrielalao** [comment]
> @gabrielalao Looks like the iOS build is still failing

Yes, I researched this reason, but seems like react native module has an issue for this. I think the only way to eliminate this issue is to upgrade expo and react native as the latest one.

**Comment by tjulien** [comment]
@gabrielalao I see, so we may need to do that? This was working on master, did something change?

**Comment by gabrielalao** [comment]
@tjulien  Expo 40 was not going to build on my pc and I didn't met the build issue before. Expo 40 is deprecated and have some issues, so I upgrade to 43.

**Comment by gabrielalao** [comment]
> I didn't change anything on this branch before upgrade, only update the code not change any gradle or podfile. but the build was failed, weird

**Comment by gabrielalao** [comment]
I will look into circleci error message again.

**Comment by gabrielalao** [comment]
@tjulien All checks have passed now. We can merge to master branch and release

File: example/android/app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -1,33 +1,22 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android" package="io.radar.example">
   <uses-permission android:name="android.permission.INTERNET"/>
+  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
   <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
-  <uses-permission android:name="android.permission.READ_PHONE_STATE"/>
```
Comment (tjulien) [inline example/android/app/src/main/AndroidManifest.xml:4]: @gabrielalao some of these permissions we may not need.  But we do need things related to location, bluetooth and network.

Comment (gabrielalao) [reply example/android/app/src/main/AndroidManifest.xml:4]: @tjulien gotcha, I will check permission again

File: package.json [hunk 7]
```diff
@@ -3,7 +3,7 @@
   "description": "React Native module for Radar, the leading geofencing and location tracking platform",
   "homepage": "https://radar.com",
   "license": "Apache-2.0",
-  "version": "3.5.6",
+  "version": "3.5.7",
```
Comment (tjulien) [inline package.json:6]: We need to run npm install so that this gets reflected in package-lock.json

Comment (gabrielalao) [reply package.json:6]: ok

File: example/android/build.gradle [hunk 8]
```diff
@@ -2,30 +2,42 @@
 
 buildscript {
     ext {
-        buildToolsVersion = '30.0.2'
+        buildToolsVersion = "29.0.3"
```
Comment (tjulien) [inline example/android/build.gradle:5]: Interesting, downgrading?  expo version thing?  related to fixing the android build?

Comment (gabrielalao) [reply example/android/build.gradle:5]: Yes, I created again android/ios project using `expo prebuild` and that version number is prefilled by expo cli.

**Comment by gabrielalao** [comment]
Checks have passed.

File: example/package.json [hunk 9]
```diff
@@ -1,32 +1,34 @@
...
+    "react-native-web": "0.17.1",
+    "react-native-safe-area-context": "3.3.2"
   },
   "devDependencies": {
-    "@babel/core": "~7.9.0",
+    "@babel/core": "^7.12.9",
     "babel-jest": "~25.2.6",
     "jest": "~25.2.6",
+    "radar-sdk-js": "^3.3.0",
     "react-test-renderer": "~16.13.1",
```
Comment (tjulien) [inline example/package.json:30]: I'm getting this on my machine on this branch:

```
tim@tjulien example % npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: example@1.0.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.13.1" from react-test-renderer@16.13.1
npm ERR! node_modules/react-test-renderer
npm ERR!   dev react-test-renderer@"~16.13.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/tim/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tim/.npm/_logs/2022-11-18T04_17_22_095Z-debug.log
```

Comment (gabrielalao) [reply example/package.json:30]: pls use `yarn` command instead of `npm install`

**Comment by tjulien** [comment]
example is working for me on iOS. Android is running in the simulator, the app loads, but the application logic in App.js does not seem to execute.  Do you know how to inspect logs?  I would expect a number console.logs to come out, like on iOS

**Comment by gabrielalao** [comment]
> example is working for me on iOS. Android is running in the simulator, the app loads, but the application logic in App.js does not seem to execute. Do you know how to inspect logs? I would expect a number console.logs to come out, like on iOS

could you run those commands?
1. `npm run android` to build the app and install on android
2. `npm start` to open the metro, and in this terminal you could see the logs
3.  if you want to see the logs on chrome browser, pls type key `d`,  this will open popup on android emulator, then pls select `Debug' to open chrome debug browser.
4. and you could open `inspect` in this browser, and you could see the console message on inspector window.

**Comment by gabrielalao** [comment]
@tjulien attached screenshot for android debug

**Review by tjulien — APPROVED** [review]

**Pull request merged**